### PR TITLE
CRYP-21: Setup users db schema

### PR DESCRIPTION
**Pull request by jason-gerard**
## Changes
- Add entity for user using typeorm
- Refactor modules to allow for orm integration
- Implement config file and process injection

## Dev notes
- Updated dockerfile for api because the slim base image did not include some system level libraries required for hot reload on macos and linux (ubuntu)

### CODE REVIEW

File: packages/common/src/entities/user.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import { Column, CreateDateColumn, Entity, PrimaryGeneratedColumn } from "typeorm";
+
+@Entity()
+export class User {
```
Comment (alexbara2000) [inline packages/common/src/entities/user.ts:4]: So would this be 1 table in the database? Would we need to do the same concept for all the different entities that go in the db?

**Review by alexbara2000 — APPROVED** [review]

File: packages/common/src/entities/user.ts (hunk 1 above)
Comment (jason-gerard) [reply packages/common/src/entities/user.ts:4]: Yep and every entity we have in our system i.e. wallet, transaction, etc will have a file for them. Later on we will have many to many joins between our tables that will create a new join table without an explicit file.

File: packages/api/src/base/base.controller.ts [hunk 2]
```diff
@@ -1,5 +1,5 @@
 import { Controller, Get } from "@nestjs/common";
-import { BaseService } from "../../services/base/base.service";
+import { BaseService } from "./base.service";
```
Comment (Andre-Ibrahim) [inline packages/api/src/base/base.controller.ts:2]: why did you remove the folders for services and controllers?

**Review by Andre-Ibrahim — APPROVED** [review]

File: packages/api/src/base/base.controller.ts (hunk 2 above)
Comment (jason-gerard) [reply packages/api/src/base/base.controller.ts:2]: This is a more domain driven design approach of setting up the directories. Instead of splitting horizontally by layers we split vertically by resources / entities (the entity itself is stored in the shared common package though).

NestJS recommends it as well, if you don't set it up like this you have to do some weird stuff to get the repository code gen to work and I also wanted to try it out to see how we like it, if its ends up being worse its always an easy refactor.

Comment (Andre-Ibrahim) [reply packages/api/src/base/base.controller.ts:2]: yea i found it odd that nest setup didn't have directories but your right should be a quick fix if we don't like it

Comment (Andre-Ibrahim) [reply packages/api/src/base/base.controller.ts:2]: merge it
dew it!
